src/main/js/controls/Bar/helpers/legendHelpers.js

```diff
@@ -47,16 +47,55 @@ const clickHandler = (graphContext, control, config, canvasSVG) => (
             shownTargets.push(item.key);
         }
     };
+    if (config.shownTargets.length === 0) {
+        graphContext.contentKeys.forEach((i) => {
+            canvasSVG
+                .selectAll(`.${styles.taskBar}[aria-describedby="${i}"]`)
+                .attr("aria-hidden", isLegendSelected(d3.select(element)));
+        });
+    }
     legendClickHandler(element);
     updateShownTarget(config.shownTargets, item);
-    canvasSVG
-        .selectAll(`g[aria-describedby="${item.key}"]`)
-        .attr("aria-hidden", isLegendSelected(d3.select(element)));
-    setSelectionIndicatorAttributes(
-        canvasSVG.selectAll(`.${styles.taskBarSelection}`),
-        false
-    );
-    window.requestAnimationFrame(onAnimationHandler(graphContext));
+    if (config.shownTargets.length >= 1) {
+        canvasSVG
+            .selectAll(`g[aria-describedby="${item.key}"]`)
+            .attr("aria-hidden", isLegendSelected(d3.select(element)));
+        canvasSVG
+            .selectAll(`.${styles.taskBar}[aria-describedby="${item.key}"]`)
+            .attr("aria-hidden", isLegendSelected(d3.select(element)));
+        canvasSVG
+            .selectAll(
+                `.${styles.region}[aria-describedby="region_${item.key}"]`
+            )
+            .attr("aria-hidden", function () {
+                return !(d3.select(this).attr("aria-hidden") === "true");
+            });
+        setSelectionIndicatorAttributes(
+            canvasSVG.selectAll(`.${styles.taskBarSelection}`),
+            false
+        );
+        setSelectionIndicatorAttributes(
+            canvasSVG.selectAll(`.${styles.taskBarSelection}`),
+            false
+        );
+        window.requestAnimationFrame(onAnimationHandler(graphContext));
+    } else {
```

Comment:
Doesn't this `else` block cover the same condition as the initial `if(config.shownTargets.length === 0)`?  Can these two blocks be consolidated?

Reply:
Yes but that if statement (if(config.shownTargets.length === 0)) needs to get executed before the on click execution gets started. That is the reason why written that separately.

Reply:
All the above code are present in the `clickHandler` function

- How does if statement (if(config.shownTargets.length === 0)) executed before we click on the onclick?

- When will the else statement execute?